### PR TITLE
Add xmlutils module

### PR DESCRIPTION
**Pull request by pramodya1994**
```ballerina
import ballerina/xmlutils;

function testToJson() returns json {
    var x1 = xml `<!-- outer comment -->`;
    var x2 = xml `<name>supun</name>`;
    xml x3 = x1 + x2;
    json j = xmlutils:toJSON(x3);
    return j;
}
```
j will be `{ name: "supun" }`

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Converts a XML to the corresponding JSON representation.
+ *
+ * @since 0.90
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "xmlutils",
+        functionName = "toJSON",
+        args = {@Argument(name = "options", type = TypeKind.RECORD, structType = "Options",
```
Comment (ldclakmal) [inline stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]:
```suggestion
```

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                structPackage = "ballerina.lang.xmls")},
```
Comment (ldclakmal) [inline stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]:
```suggestion
```

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        returnType = {@ReturnType(type = TypeKind.JSON)},
```
Comment (ldclakmal) [inline stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]:
```suggestion
```

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ErrorHandler.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+     * Handle any xpath related exception.
+     *
+     * @param operation     Operation that executed
+     * @param e             Throwable to handle
+     */
+    public static void handleXMLException(String operation, Throwable e) {
+        // here local message of the cause is logged whenever possible, to avoid java class being logged
+        // along with the error message.
+        if (e instanceof BallerinaException && ((BallerinaException) e).getDetail() != null) {
+            throw new BallerinaException(BallerinaErrorReasons.XML_OPERATION_ERROR, "Failed to " + operation + ": " +
```
Comment (wggihan) [inline stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ErrorHandler.java]: Let's throw ErrorValue instead of BallerinaException.

Comment (pramodya1994) [reply stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ErrorHandler.java]: Removed this file and used `handleXMLException` method in `BLangExceptionHelper` class.

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java (hunk 1 above)
Comment (pramodya1994) [reply stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]: Done

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java (hunk 2 above)
Comment (pramodya1994) [reply stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]: Done

File: stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java (hunk 3 above)
Comment (pramodya1994) [reply stdlib/xmlutils/src/main/java/org/ballerinalang/stdlib/xmlutils/ToJSON.java]: Done

File: stdlib/xmlutils/src/main/ballerina/src/xmlutils/xml-to-json.bal [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Converts a XML object to a JSON representation.
+#
+# + x - The xml source
+# + options - xmlOptions struct for XML to JSON conversion properties
+# + return - JSON representation of the given XML
+public function toJSON(xml x, record {
```
Comment (ldclakmal) [inline stdlib/xmlutils/src/main/ballerina/src/xmlutils/xml-to-json.bal]: Shall we update the API as follows? This enables the end-user to provide the XML only without passing the options record in prominent scenarios.
```suggestion
public type XmlOptions record {
    string attributePrefix = "@";
    boolean preserveNamespaces = true;
}
public function toJSON(xml x, XmlOptions options = {}) returns json|error = external;
```

File: stdlib/xmlutils/src/test/resources/test-src/xml-to-json-test.bal [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/xmlutils;
+
+function testToJson() returns json|error {
+    var x1 = xml `<!-- outer comment -->`;
+    var x2 = xml `<name>supun</name>`;
+    xml x3 = x1 + x2;
+    json|error j = xmlutils:toJSON(x3, {});
```
Comment (ldclakmal) [inline stdlib/xmlutils/src/test/resources/test-src/xml-to-json-test.bal]:
```suggestion
    json|error j = xmlutils:toJSON(x3);
```

Comment (ldclakmal) [reply stdlib/xmlutils/src/test/resources/test-src/xml-to-json-test.bal]: This applies iff https://github.com/ballerina-platform/ballerina-lang/pull/18436/files#r320624993 is okay.

File: stdlib/xmlutils/src/main/ballerina/src/xmlutils/xml-to-json.bal (hunk 5 above)
Comment (pramodya1994) [reply stdlib/xmlutils/src/main/ballerina/src/xmlutils/xml-to-json.bal]: Added

File: stdlib/xmlutils/src/test/resources/test-src/xml-to-json-test.bal (hunk 6 above)
Comment (pramodya1994) [reply stdlib/xmlutils/src/test/resources/test-src/xml-to-json-test.bal]: Added

File: stdlib/xmlutils/build.gradle [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+dependencies {
+    baloImplementation project(path: ':ballerina-lang:annotations', configuration: 'baloImplementation')
+    baloImplementation project(path: ':ballerina-runtime-api', configuration: 'baloImplementation')
+
+    baloCreat project(':lib-creator')
+    implementation project(':ballerina-lang')
+    implementation project(':ballerina-runtime')
+    implementation project(':ballerina-logging')
```
Comment (SupunS) [inline stdlib/xmlutils/build.gradle]: do we need this?

File: stdlib/xmlutils/build.gradle [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    implementation project(':ballerina-runtime-api')
+
+    testCompile project(path: ':ballerina-test-common', configuration: 'tests')
+    testCompile 'org.testng:testng'
+    testCompile 'org.slf4j:slf4j-jdk14'
+
+    testCompile project(path: ':ballerina-test-utils', configuration: 'shadow')
+    testCompile project(':ballerina-test-utils')
+    testCompile project(':ballerina-reflect')
```
Comment (SupunS) [inline stdlib/xmlutils/build.gradle]: need this dependency?

File: stdlib/xmlutils/build.gradle (hunk 7 above)
Comment (pramodya1994) [reply stdlib/xmlutils/build.gradle]: removed

File: stdlib/xmlutils/build.gradle (hunk 8 above)
Comment (pramodya1994) [reply stdlib/xmlutils/build.gradle]: removed

**Pull request merged**